### PR TITLE
Issue 1314b

### PR DESCRIPTION
**Pull request by chipfreitag**
Fixes for ROCm MDRange.

### CODE REVIEW

**Comment by ibaned** [comment]
got some CUDA test failures when I ran a spot check on this:
```
make[2]: Leaving directory `/home/daibane/src/kokkos/testing/TestAll_2018-04-16_09.38.27/clang/6.0/Cuda_Pthread-release/core/unit_test'
make -f /home/daibane/src/kokkos/core/unit_test/Makefile KOKKOS_SRC_PATH=/home/daibane/src/kokkos CXX=/home/projects/x86-64/clang/6.0/bin/clang++ KOKKOS_DEVICES=Cuda_Pthread KOKKOS_ARCH=SNB,Volta70 CXXFLAGS="-O3 -Wall -Wshadow -pedantic -Wsign-compare -Wtype-limits -Wuninitialized " LDFLAGS="-O3 " GTEST_PATH=/home/daibane/src/kokkos/tpls/gtest KOKKOS_PATH=/home/daibane/src/kokkos test
make[2]: Entering directory `/home/daibane/src/kokkos/testing/TestAll_2018-04-16_09.38.27/clang/6.0/Cuda_Pthread-release/core/unit_test'
./KokkosCore_UnitTest_Cuda
[==========] Running 114 tests from 3 test cases.
[----------] Global test environment set-up.
[----------] 64 tests from cuda
[ RUN      ] cuda.init
[       OK ] cuda.init (0 ms)
[ RUN      ] cuda.dispatch
[       OK ] cuda.dispatch (59 ms)
[ RUN      ] cuda.compiler_macros
[       OK ] cuda.compiler_macros (0 ms)
[ RUN      ] cuda.policy_construction
[       OK ] cuda.policy_construction (0 ms)
[ RUN      ] cuda.impl_shared_alloc
[       OK ] cuda.impl_shared_alloc (76 ms)
[ RUN      ] cuda.range_for
[       OK ] cuda.range_for (2 ms)
[ RUN      ] cuda.range_reduce
[       OK ] cuda.range_reduce (2 ms)
[ RUN      ] cuda.range_scan
[       OK ] cuda.range_scan (1 ms)
[ RUN      ] cuda.view_api
[       OK ] cuda.view_api (15 ms)
[ RUN      ] cuda.view_remap
[       OK ] cuda.view_remap (2 ms)
[ RUN      ] cuda.view_mirror_nonconst
[       OK ] cuda.view_mirror_nonconst (0 ms)
[ RUN      ] cuda.view_stride_method
[       OK ] cuda.view_stride_method (0 ms)
[ RUN      ] cuda.anonymous_space
[       OK ] cuda.anonymous_space (0 ms)
[ RUN      ] cuda.view_mapping
[       OK ] cuda.view_mapping (79 ms)
[ RUN      ] cuda.view_mapping_operator
[       OK ] cuda.view_mapping_operator (13 ms)
[ RUN      ] cuda.view_mapping_atomic
[       OK ] cuda.view_mapping_atomic (1 ms)
[ RUN      ] cuda.view_mapping_class_value
[       OK ] cuda.view_mapping_class_value (0 ms)
[ RUN      ] cuda.view_mapping_assignable
[       OK ] cuda.view_mapping_assignable (0 ms)
[ RUN      ] cuda.view_mapping_subview
[       OK ] cuda.view_mapping_subview (2 ms)
[ RUN      ] cuda.view_nested_view
[       OK ] cuda.view_nested_view (0 ms)
[ RUN      ] cuda.functor_analysis
[       OK ] cuda.functor_analysis (0 ms)
[ RUN      ] cuda.long_reduce
[       OK ] cuda.long_reduce (16 ms)
[ RUN      ] cuda.double_reduce
[       OK ] cuda.double_reduce (16 ms)
[ RUN      ] cuda.reducers
[       OK ] cuda.reducers (13 ms)
[ RUN      ] cuda.long_reduce_dynamic
[       OK ] cuda.long_reduce_dynamic (17 ms)
[ RUN      ] cuda.double_reduce_dynamic
[       OK ] cuda.double_reduce_dynamic (17 ms)
[ RUN      ] cuda.long_reduce_dynamic_view
[       OK ] cuda.long_reduce_dynamic_view (19 ms)
[ RUN      ] cuda.reduction_deduction
[       OK ] cuda.reduction_deduction (0 ms)
[ RUN      ] cuda.scan
TestScan(129,6370) != 8515
TestScan(193,14162) != 18915
TestScan(387,32980) != 75466
[       OK ] cuda.scan (387 ms)
[ RUN      ] cuda.complex_construction
[       OK ] cuda.complex_construction (1 ms)
[ RUN      ] cuda.complex_basic_math
[       OK ] cuda.complex_basic_math (0 ms)
[ RUN      ] cuda.complex_special_funtions
[       OK ] cuda.complex_special_funtions (1 ms)
[ RUN      ] cuda.atomic_operations
[       OK ] cuda.atomic_operations (233 ms)
[ RUN      ] cuda.atomic_views_integral
[       OK ] cuda.atomic_views_integral (14935 ms)
[ RUN      ] cuda.atomic_views_nonintegral
[       OK ] cuda.atomic_views_nonintegral (8803 ms)
[ RUN      ] cuda.atomic_view_api
[       OK ] cuda.atomic_view_api (1 ms)
[ RUN      ] cuda.atomics
[       OK ] cuda.atomics (199 ms)
[ RUN      ] cuda.team_for
[       OK ] cuda.team_for (2 ms)
[ RUN      ] cuda.team_reduce
/home/daibane/src/kokkos/core/unit_test/TestTeam.hpp:151: Failure
Value of: size_t( total )
  Actual: 3724386434828339393
Expected: size_t( ( N - 1 ) * ( N ) ) / 2
Which is: 130816
/home/daibane/src/kokkos/core/unit_test/TestTeam.hpp:151: Failure
Value of: size_t( total )
  Actual: 3582128929547040257
Expected: size_t( ( N - 1 ) * ( N ) ) / 2
Which is: 130816
/home/daibane/src/kokkos/core/unit_test/TestTeam.hpp:151: Failure
Value of: size_t( total )
  Actual: 10416509546753368797
Expected: size_t( ( N - 1 ) * ( N ) ) / 2
Which is: 32767872000
/home/daibane/src/kokkos/core/unit_test/TestTeam.hpp:151: Failure
Value of: size_t( total )
  Actual: 780597421294173
Expected: size_t( ( N - 1 ) * ( N ) ) / 2
Which is: 32767872000
[  FAILED  ] cuda.team_reduce (2 ms)
[ RUN      ] cuda.team_shared_request
[       OK ] cuda.team_shared_request (0 ms)
[ RUN      ] cuda.team_scratch_request
[       OK ] cuda.team_scratch_request (0 ms)
[ RUN      ] cuda.team_lambda_shared_request
[       OK ] cuda.team_lambda_shared_request (0 ms)
[ RUN      ] cuda.shmem_size
[       OK ] cuda.shmem_size (0 ms)
[ RUN      ] cuda.multi_level_scratch
[       OK ] cuda.multi_level_scratch (186 ms)
[ RUN      ] cuda.team_long_reduce
[       OK ] cuda.team_long_reduce (34 ms)
[ RUN      ] cuda.team_double_reduce
[       OK ] cuda.team_double_reduce (33 ms)
[ RUN      ] cuda.template_meta_functions
[       OK ] cuda.template_meta_functions (1 ms)
[ RUN      ] cuda.view_aggregate
[       OK ] cuda.view_aggregate (0 ms)
[ RUN      ] cuda.memory_pool
[       OK ] cuda.memory_pool (9 ms)
[ RUN      ] cuda.cxx11
CXX11 ( test = 'ReduceTest TeamPolicy' FAILED : 8.01522e-303 != 1.23868e-302
/home/daibane/src/kokkos/core/unit_test/TestCXX11.hpp:354: Failure
Value of: ( TestCXX11::Test< TEST_EXECSPACE >( 4 ) )
  Actual: false
Expected: true
[  FAILED  ] cuda.cxx11 (1 ms)
[ RUN      ] cuda.tile_layout
[       OK ] cuda.tile_layout (6 ms)
[ RUN      ] cuda.viewctorprop_embedded_dim
[       OK ] cuda.viewctorprop_embedded_dim (1 ms)
[ RUN      ] cuda.mdrange_for
[       OK ] cuda.mdrange_for (34 ms)
[ RUN      ] cuda.mdrange_reduce
[       OK ] cuda.mdrange_reduce (26 ms)
[ RUN      ] cuda.mdrange_array_reduce
[       OK ] cuda.mdrange_array_reduce (1 ms)
[ RUN      ] cuda.mdrange_neg_idx
[       OK ] cuda.mdrange_neg_idx (2 ms)
[ RUN      ] cuda.crs_count_fill
[       OK ] cuda.crs_count_fill (5 ms)
[ RUN      ] cuda.workgraph_fib
[       OK ] cuda.workgraph_fib (742 ms)
[ RUN      ] cuda.space_access
[       OK ] cuda.space_access (0 ms)
[ RUN      ] cuda.uvm
[       OK ] cuda.uvm (1 ms)
[ RUN      ] cuda.uvm_num_allocs
[       OK ] cuda.uvm_num_allocs (12749 ms)
[ RUN      ] cuda.impl_view_accessible
[       OK ] cuda.impl_view_accessible (7 ms)
[ RUN      ] cuda.impl_view_texture
[       OK ] cuda.impl_view_texture (1 ms)
[ RUN      ] cuda.unique_token
TestUniqueToken max reuse = 3843
[       OK ] cuda.unique_token (215 ms)
[----------] 64 tests from cuda (38971 ms total)

[----------] 38 tests from cuda_uvm
[ RUN      ] cuda_uvm.impl_shared_alloc
[       OK ] cuda_uvm.impl_shared_alloc (82 ms)
[ RUN      ] cuda_uvm.view_api
[       OK ] cuda_uvm.view_api (32 ms)
[ RUN      ] cuda_uvm.view_remap
[       OK ] cuda_uvm.view_remap (1 ms)
[ RUN      ] cuda_uvm.view_mirror_nonconst
[       OK ] cuda_uvm.view_mirror_nonconst (1 ms)
[ RUN      ] cuda_uvm.view_stride_method
[       OK ] cuda_uvm.view_stride_method (0 ms)
[ RUN      ] cuda_uvm.anonymous_space
[       OK ] cuda_uvm.anonymous_space (0 ms)
[ RUN      ] cuda_uvm.view_mapping
[       OK ] cuda_uvm.view_mapping (80 ms)
[ RUN      ] cuda_uvm.view_mapping_operator
[       OK ] cuda_uvm.view_mapping_operator (14 ms)
[ RUN      ] cuda_uvm.view_mapping_atomic
[       OK ] cuda_uvm.view_mapping_atomic (1 ms)
[ RUN      ] cuda_uvm.view_mapping_class_value
[       OK ] cuda_uvm.view_mapping_class_value (0 ms)
[ RUN      ] cuda_uvm.view_mapping_assignable
[       OK ] cuda_uvm.view_mapping_assignable (0 ms)
[ RUN      ] cuda_uvm.view_mapping_subview
[       OK ] cuda_uvm.view_mapping_subview (1 ms)
[ RUN      ] cuda_uvm.view_subview_auto_1d_left
[       OK ] cuda_uvm.view_subview_auto_1d_left (6 ms)
[ RUN      ] cuda_uvm.view_subview_auto_1d_right
[       OK ] cuda_uvm.view_subview_auto_1d_right (5 ms)
[ RUN      ] cuda_uvm.view_subview_auto_1d_stride
[       OK ] cuda_uvm.view_subview_auto_1d_stride (5 ms)
[ RUN      ] cuda_uvm.view_subview_assign_strided
[       OK ] cuda_uvm.view_subview_assign_strided (8 ms)
[ RUN      ] cuda_uvm.view_subview_left_0
[       OK ] cuda_uvm.view_subview_left_0 (0 ms)
[ RUN      ] cuda_uvm.view_subview_left_1
[       OK ] cuda_uvm.view_subview_left_1 (1 ms)
[ RUN      ] cuda_uvm.view_subview_left_2
[       OK ] cuda_uvm.view_subview_left_2 (0 ms)
[ RUN      ] cuda_uvm.view_subview_left_3
[       OK ] cuda_uvm.view_subview_left_3 (1 ms)
[ RUN      ] cuda_uvm.view_subview_right_0
[       OK ] cuda_uvm.view_subview_right_0 (1 ms)
[ RUN      ] cuda_uvm.view_subview_right_1
[       OK ] cuda_uvm.view_subview_right_1 (0 ms)
[ RUN      ] cuda_uvm.view_subview_right_3
[       OK ] cuda_uvm.view_subview_right_3 (1 ms)
[ RUN      ] cuda_uvm.view_subview_layoutleft_to_layoutleft
[       OK ] cuda_uvm.view_subview_layoutleft_to_layoutleft (6 ms)
[ RUN      ] cuda_uvm.view_subview_layoutright_to_layoutright
[       OK ] cuda_uvm.view_subview_layoutright_to_layoutright (4 ms)
[ RUN      ] cuda_uvm.view_subview_1d_assign
[       OK ] cuda_uvm.view_subview_1d_assign (6 ms)
[ RUN      ] cuda_uvm.view_subview_1d_assign_atomic
[       OK ] cuda_uvm.view_subview_1d_assign_atomic (5 ms)
[ RUN      ] cuda_uvm.view_subview_1d_assign_randomaccess
[       OK ] cuda_uvm.view_subview_1d_assign_randomaccess (6 ms)
[ RUN      ] cuda_uvm.view_subview_2d_from_3d
[       OK ] cuda_uvm.view_subview_2d_from_3d (87 ms)
[ RUN      ] cuda_uvm.view_subview_2d_from_3d_atomic
[       OK ] cuda_uvm.view_subview_2d_from_3d_atomic (88 ms)
[ RUN      ] cuda_uvm.view_subview_2d_from_3d_randomaccess
[       OK ] cuda_uvm.view_subview_2d_from_3d_randomaccess (89 ms)
[ RUN      ] cuda_uvm.view_subview_3d_from_5d_left
[       OK ] cuda_uvm.view_subview_3d_from_5d_left (532 ms)
[ RUN      ] cuda_uvm.view_subview_3d_from_5d_left_atomic
[       OK ] cuda_uvm.view_subview_3d_from_5d_left_atomic (529 ms)
[ RUN      ] cuda_uvm.view_subview_3d_from_5d_left_randomaccess
[       OK ] cuda_uvm.view_subview_3d_from_5d_left_randomaccess (532 ms)
[ RUN      ] cuda_uvm.view_subview_3d_from_5d_right
[       OK ] cuda_uvm.view_subview_3d_from_5d_right (379 ms)
[ RUN      ] cuda_uvm.view_subview_3d_from_5d_right_atomic
[       OK ] cuda_uvm.view_subview_3d_from_5d_right_atomic (378 ms)
[ RUN      ] cuda_uvm.view_subview_3d_from_5d_right_randomaccess
[       OK ] cuda_uvm.view_subview_3d_from_5d_right_randomaccess (378 ms)
[ RUN      ] cuda_uvm.view_test_unmanaged_subview_reset
[       OK ] cuda_uvm.view_test_unmanaged_subview_reset (1 ms)
[----------] 38 tests from cuda_uvm (3260 ms total)

[----------] 12 tests from cuda_hostpinned
[ RUN      ] cuda_hostpinned.impl_shared_alloc
[       OK ] cuda_hostpinned.impl_shared_alloc (67 ms)
[ RUN      ] cuda_hostpinned.view_api
[       OK ] cuda_hostpinned.view_api (31 ms)
[ RUN      ] cuda_hostpinned.view_remap
[       OK ] cuda_hostpinned.view_remap (1 ms)
[ RUN      ] cuda_hostpinned.view_mirror_nonconst
[       OK ] cuda_hostpinned.view_mirror_nonconst (0 ms)
[ RUN      ] cuda_hostpinned.view_stride_method
[       OK ] cuda_hostpinned.view_stride_method (0 ms)
[ RUN      ] cuda_hostpinned.anonymous_space
[       OK ] cuda_hostpinned.anonymous_space (1 ms)
[ RUN      ] cuda_hostpinned.view_mapping
[       OK ] cuda_hostpinned.view_mapping (74 ms)
[ RUN      ] cuda_hostpinned.view_mapping_operator
[       OK ] cuda_hostpinned.view_mapping_operator (4 ms)
[ RUN      ] cuda_hostpinned.view_mapping_atomic
[       OK ] cuda_hostpinned.view_mapping_atomic (1 ms)
[ RUN      ] cuda_hostpinned.view_mapping_class_value
[       OK ] cuda_hostpinned.view_mapping_class_value (0 ms)
[ RUN      ] cuda_hostpinned.view_mapping_assignable
[       OK ] cuda_hostpinned.view_mapping_assignable (0 ms)
[ RUN      ] cuda_hostpinned.view_mapping_subview
[       OK ] cuda_hostpinned.view_mapping_subview (1 ms)
[----------] 12 tests from cuda_hostpinned (180 ms total)

[----------] Global test environment tear-down
[==========] 114 tests from 3 test cases ran. (42411 ms total)
[  PASSED  ] 112 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] cuda.team_reduce
[  FAILED  ] cuda.cxx11

 2 FAILED TESTS
make[2]: *** [test-cuda] Error 1
make[2]: Leaving directory `/home/daibane/src/kokkos/testing/TestAll_2018-04-16_09.38.27/clang/6.0/Cuda_Pthread-release/core/unit_test'
make[1]: *** [test] Error 2
make[1]: Leaving directory `/home/daibane/src/kokkos/testing/TestAll_2018-04-16_09.38.27/clang/6.0/Cuda_Pthread-release/core/unit_test'
make: *** [test] Error 2
  Starting job cuda-9.1-Cuda_OpenMP-release
```

File: core/src/KokkosExp_MDRangePolicy.hpp [hunk 1]
```diff
@@ -376,7 +417,7 @@ struct MDRangePolicy
               || ((int)inner_direction == (int)Left && (i > 0)) )
           {
             if ( m_prod_tile_dims < 256 ) {
-              m_tile[i] = 2;
+              m_tile[i] = 4;
```
Comment (crtrott) [inline core/src/KokkosExp_MDRangePolicy.hpp]: This might be the cause of CUDA test failures

**Comment by crtrott** [comment]
I am checking with the change now.

**Comment by crtrott** [comment]
Rerunning SpotCheck. The failure was because of issue #1534 and the fix not being merged into this branch yet.

**Comment by crtrott** [comment]
```
Running on machine: apollo
Repository Status:  1b360c1da3c8d122a52f90ea67e099ce3781f3f6 Merge branch 'develop' into issue-1314b


Going to test compilers:  gcc/4.8.4 gcc/5.3.0 intel/16.0.1 clang/3.9.0 clang/6.0 cuda/9.1
Testing compiler gcc/4.8.4
  Starting job gcc-4.8.4-OpenMP-release
  PASSED gcc-4.8.4-OpenMP-release
Testing compiler gcc/5.3.0
  Starting job gcc-4.8.4-Pthread-release
  PASSED gcc-4.8.4-Pthread-release
Testing compiler intel/16.0.1
  Starting job gcc-5.3.0-Serial-release
  PASSED gcc-5.3.0-Serial-release
Testing compiler clang/3.9.0
  Starting job intel-16.0.1-OpenMP-release
  PASSED intel-16.0.1-OpenMP-release
Testing compiler clang/6.0
  Starting job clang-3.9.0-Pthread_Serial-release
  PASSED clang-3.9.0-Pthread_Serial-release
Testing compiler cuda/9.1
  Starting job clang-6.0-Cuda_Pthread-release
  PASSED clang-6.0-Cuda_Pthread-release
  Starting job cuda-9.1-Cuda_OpenMP-release
  PASSED cuda-9.1-Cuda_OpenMP-release
#######################################################
PASSED TESTS
#######################################################
clang-3.9.0-Pthread_Serial-release build_time=117 run_time=416
clang-6.0-Cuda_Pthread-release build_time=212 run_time=425
cuda-9.1-Cuda_OpenMP-release build_time=262 run_time=258
gcc-4.8.4-OpenMP-release build_time=99 run_time=135
gcc-4.8.4-Pthread-release build_time=91 run_time=286
gcc-5.3.0-Serial-release build_time=94 run_time=178
intel-16.0.1-OpenMP-release build_time=316 run_time=168
```

**Pull request merged**